src/Workspaces/Core/Portable/Classification/ClassificationContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Classification
+{
+    /// <summary>
+    /// A context for aggregating classifications from one or more <see cref="ClassificationProvider"/>'s.
+    /// </summary>
+    internal class ClassificationContext
+    {
+        private readonly List<ClassifiedSpan> _spans;
+        private readonly HashSet<ClassifiedSpan> _spanSet;
+
+        internal ClassificationContext(List<ClassifiedSpan> classifiedSpans, HashSet<ClassifiedSpan> spanSet)
+        {
+            _spans = classifiedSpans;
+            _spanSet = spanSet;
+        }
+
+        public void AddClassification(ClassifiedSpan classifiedSpan)
```

Comment:
I like that idea! However, the Classifier API is already public and has been so since 1.0. ClassifiedSpan itself is public right here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Classification/ClassifiedSpan.cs,9348503c8c058a4b.